src/backend/Level0/CHIPBackendLevel0.cc

```diff
@@ -555,7 +556,8 @@ void CHIPStaleEventMonitorLevel0::monitor() {
         // Add the most course-grain lock here in case event destructor is not
         // thread-safe
         std::lock_guard Lock(Backend->Mtx);
-        delete E;
+        if (!E->EventPool) // Delete event if not owned by an event pool.
+          delete E;
```

Comment:
This should probably go away entirely. Event creation/destruction is managed by pools now. all other events are managed by the user